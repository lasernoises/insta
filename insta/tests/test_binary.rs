use std::io::Write;

#[test]
fn test_binary_snapshot() {
    insta::assert_binary_snapshot!("txt", |file| {
        file.write_all(b"test").unwrap();
    });
}

#[test]
fn test_file_extension_collision() {
    // ubuntu snap packages also have the .snap extension so let's make sure that doesn't cause
    // problems
    insta::assert_binary_snapshot!("snap", |file| {
        file.write_all(b"test").unwrap();
    });
}

#[test]
fn test_file_empty_extension() {
    insta::assert_binary_snapshot!("", |file| {
        file.write_all(b"test").unwrap();
    });
}
