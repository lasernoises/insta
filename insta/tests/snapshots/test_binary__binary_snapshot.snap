---
source: insta/tests/test_binary.rs
expression: "|file| { file.write_all(b\"test\").unwrap(); }"
extension: txt
snapshot_type: binary
---
